configure.ac

```diff
@@ -198,21 +198,36 @@ saved_user_CPPFLAGS="$CPPFLAGS"
 ##
 ##  Regex:
 ##
-##      -Werror         Literal -Werror
-##      \(              Start optional capturing group
-##      =               Literal equals sign
-##      [^[:space:]-]   Non-space characters
-##      \+              1 or more of the above
-##      \)              End optional capturing group
-##      \?              0 or 1 capturing group matches
+##      -Werror        Literal -Werror
+##      \(             Start optional capturing group
+##      =              Literal equals sign
+##      [^[:space:]]   Non-space characters
+##      \+             1 or more of the above
+##      \)             End optional capturing group
+##      \?             0 or 1 capturing group matches
 ##
-WERROR_SED= "sed -e 's/-Werror\(=[^[:space:]]\+\)\?//g'"
-
-CFLAGS="`echo $CFLAGS | $WERROR_SED`"
-CXXFLAGS="`echo $CXXFLAGS | $WERROR_SED`"
-FCFLAGS="`echo $FCFLAGS | $WERROR_SED`"
-JAVACFLAGS="`echo $JAVACFLAGS | $WERROR_SED`"
-CPPFLAGS="`echo $CPPFLAGS | $WERROR_SED`"
+## Note that the outer pair of '[]' ends up getting removed
+WERROR_SED='s/-Werror\(=[[^[:space:]]]\+\)\?//g'
+CFLAGS_SED="`echo $CFLAGS | sed -e $WERROR_SED`"
+if test $? -eq 0; then
```

Comment:
Check to make sure the sed command didn't fail before assigning the result back to `CFLAGS`, etc. This may cause "-Werror" to not be removed on some systems if they end up not eating the extra outer pair of `[]`, but at least it won't wipe out `CFLAGS` and others